testsuite/smoke-test/src/state_sync.rs

```diff
@@ -1008,3 +1005,118 @@ async fn verify_pruning_metrics_after_fast_sync(
         assert!(ledger_pruner_version > 0);
     }
 }
+
+/// A test method that verifies that a validator can sync after a data wipe
+/// and begin to participate in consensus. If `epoch_changes` are enabled
+/// then epoch changes can occur during test execution and validator syncing.
+/// If `fast_sync` is true, then the validator will use snapshot (fast)
+/// syncing. Otherwise, it will use the default syncing method.
+async fn verify_validator_bootstrap_and_participate(fast_sync: bool, epoch_changes: bool) {
+    // Create a swarm of 4 validators
+    let num_validators = 4;
+    let mut swarm = SwarmBuilder::new_local(num_validators)
+        .with_aptos()
+        .with_init_config(Arc::new(move |_, config, _| {
+            if fast_sync {
+                // Set the bootstrapping mode to fast syncing
+                config.state_sync.state_sync_driver.bootstrapping_mode =
+                    BootstrappingMode::DownloadLatestStates;
+                config.state_sync.storage_service.max_state_chunk_size = 30;
+            }
+        }))
+        .with_init_genesis_config(Arc::new(|genesis_config| {
+            // Shorten the required proposer history to speed up the test
+            let consensus_config = match genesis_config.consensus_config.clone() {
+                OnChainConsensusConfig::V1(consensus_config) => consensus_config,
+                OnChainConsensusConfig::V2(consensus_config) => consensus_config,
+                config => unimplemented!(
+                    "This test requires a V1 or V2 consensus config, but got: {:?}",
+                    config
+                ),
+            };
+            let leader_reputation_type = match &consensus_config.proposer_election_type {
+                ProposerElectionType::LeaderReputation(leader_reputation_type) => {
+                    leader_reputation_type
+                },
+                proposer_election_type => panic!(
+                    "This test requires a leader reputation proposer election, but got: {:?}",
+                    proposer_election_type
+                ),
+            };
+            let proposer_and_voter_config = match &leader_reputation_type {
+                LeaderReputationType::ProposerAndVoterV2(proposer_and_voter_config) => {
+                    proposer_and_voter_config
+                },
+                leader_reputation_type => panic!(
+                    "This test requires a proposer and voter V2 leader reputation, but got: {:?}",
+                    leader_reputation_type
+                ),
+            };
+            genesis_config.consensus_config = OnChainConsensusConfig::V1(ConsensusConfigV1 {
+                proposer_election_type: ProposerElectionType::LeaderReputation(
+                    LeaderReputationType::ProposerAndVoter(ProposerAndVoterConfig {
+                        proposer_window_num_validators_multiplier: 0,
```

Comment:
FYI @igor-aptos -- I'm not sure if these values make sense. But, the test does seem to pass 😄